### PR TITLE
feat: add mood tracking

### PR DESCRIPTION
**Pull request by djaiss**
Three new things:

- a widget on the dashboard
- a revamped Reports page
- a new reports for mood tracking, that we'll improve in the future of course

### CODE REVIEW

File: app/Domains/Contact/ManageMoodTrackingEvents/Web/Controllers/ContactMoodTrackingEventsController.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        $data = [
+            'account_id' => Auth::user()->account_id,
+            'author_id' => Auth::id(),
+            'vault_id' => $vaultId,
+            'contact_id' => $contactId,
+            'mood_tracking_parameter_id' => $request->input('parameter_id'),
+            'rated_at' => $carbonDate,
+            'note' => $request->input('note') ?? null,
+            'number_of_hours_slept' => $request->input('hours') ?? null,        ];
```
Comment (asbiin) [inline app/Domains/Contact/ManageMoodTrackingEvents/Web/Controllers/ContactMoodTrackingEventsController.php]: consistency:
```suggestion
            'number_of_hours_slept' => $request->input('hours') ?? null,
        ];
```

File: resources/js/Shared/Modules/FeedItems/MoodTrackingEvent.vue [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    </div>
+
+    <div class="px-3 py-2">
+      <!-- the mood tracking event still exists in the system -->
+      <div v-if="data.mood_tracking_event.object" class="flex items-center">
+        <span class="mr-2">{{ data.mood_tracking_event.description }}</span>
+        <span class="mr-2"><span class="text-gray-500">for</span> {{ data.mood_tracking_event.object.rated_at }}</span>
+        <span v-if="data.mood_tracking_event.object.number_of_hours_slept"
+          >and slept {{ data.mood_tracking_event.object.number_of_hours_slept }} hours</span
+        >
```
Comment (asbiin) [inline resources/js/Shared/Modules/FeedItems/MoodTrackingEvent.vue]: This will be too complicated to translate.

**Review by asbiin — APPROVED** [review]
ok cool

File: resources/js/Shared/Modules/FeedItems/MoodTrackingEvent.vue (hunk 2 above)
Comment (djaiss) [reply resources/js/Shared/Modules/FeedItems/MoodTrackingEvent.vue]: Very. I agree.

Comment (djaiss) [reply resources/js/Shared/Modules/FeedItems/MoodTrackingEvent.vue]: fixed

**Pull request merged**